src/common/fs_types.cc

```diff
@@ -74,6 +74,20 @@ void file_layout_t::to_legacy(ceph_file_layout *fl) const
     fl->fl_pg_pool = 0;
 }
 
+void file_layout_t::bound_encode(size_t& p, uint64_t features) const
+{
+  if (features & CEPH_FEATURE_FS_FILE_LAYOUT_V2) {
```

Comment:
I can recall investigating a problem in BlueStore when `bound_encode()` was pretty costly. The reason was in run-time checks (a load, jumping...) aggregated over the nested structures. This can become pretty painful especially when there is iteration/container somewhere along the path.

The simpler the logic for the size bound, the better. Ideally it would be just adding a constant.

Reply:
Once we pull up to C++20 it might be worth looking into using consteval for bound_encode. Anything that can't be written constexprly just won't be able to fulfill the requirement.

Reply:
haven't checked the generated assembly, but i *think* the compiler should be able to do constant folding in most cases as long as it's allowed.